### DISCUSSION TITLE
Can I use libgit2 in an app on the macOS App Store?

### DISCUSSION THREAD

**Discussion by MMcKester** [libgit2/libgit2#5657] [General] [answered]
Hi everyone,

this post is addressed to the creators of libgit2 because I am interested what your position is on libgit2 being used in apps that are distributed on the App Store. I know that there are a variety of issues raised on the web in regards to this topic, but noone really has an answer on this. And at the end it's up to the creators of this library what their standpoint is on this, especially with certain additional exceptions to the GPL license. Any feedback is highly appreciated!

Thanks a lot!

**Comment by ethomson** [comment, marked as answer]
There's at least two already.  What are these issues that are raised on the web?

**Reply by MMcKester** [reply]
Sorry I wasn't clear. GPL seems not to be compatible with the macOS App Store restrictions (so GNU GPL?). And I am developing an app using libgit2, so I am wondering if that is actually an issue or if I miss anything. The FSF did make a writeup about this topic here a while ago: https://www.fsf.org/blogs/licensing/more-about-the-app-store-gpl-enforcement

**Reply by ethomson** [reply]
Thanks for the link.  I've read it twice now.  IANAL, and this particular issue falls well outside my wheelhouse.

I would note that this library is not licensed under the GPL, it's licensed under the GPL with the linking exception (as you noted), which states:

>  In addition to the permissions in the GNU General Public License, the authors give you unlimited permission to link the compiled version of this library into combinations with other programs, and to distribute those combinations without any restriction coming from the use of this file.  (The General Public License restrictions do apply in other respects; for example, they cover modification of the file, and distribution when not linked into a combined executable.)

The goal is to permit you to use libgit2, and that if you make changes to libgit2 itself, that those changes fall under the GPL.

If you're asking for a legal interpretation of our license and how it pertains to Apple's Terms of Service, I can't give you one, since I'm not qualified to do so.  If you're asking if I care about or object to you putting an application that includes libgit2 into the app store, I do not. And if you're asking if Apple will care or object, I do not know.

Sorry if this is unhelpful, I don't have a full grasp of this problem, nor do I expect to be able to formulate one immediately, and even if I did, I would still only be comfortable giving you my opinion, not something that could be construed as legal advice.  😬

**Reply by MMcKester** [reply]
Thank you very much for your detailed answer! I will do some more investigation on my side. Again, thanks a lot!
